Retro Meeting Summary

mattkrick opened the pull request:
This completes the meeting flow.
It's probably really buggy, but I wanna get it out there for testing.
note: if you get a server error about TeamSettings, migrate down 2 clicks, then migrate back up. let me know if that appears for you, because if so, we'll have to fix the migrations.

VOTING TEST
- [ ] Go to the voting phase, can expand groups, but cannot drag cards
- [ ] can click the gray checkmark up to 3 times
- [ ] can remove votes by clicking pink checkmarks
- [ ] see updated vote info in the sidebar
- [ ] Browser B (different user) sees updated team votes remaining, but receives no user-specific info
- [ ] groups with only 1 reflection has an auto-generated title

DISCUSS TEST
- [ ] can click the discuss topic in the sidebar & navigate to that stage
- [ ] can add tasks
- [ ] team members can see the editing state & card updates & card deletions


SUMMARY TEST
- [ ] can click end meeting any time during the discuss phase
- [ ] when the meeting has ended, you go to the meeting summary
- [ ] the meeting summary shows you the tasks created, upvotied topics (does NOT distinguish between topics discussed/not discussed)
- [ ] you get an email with the summary there, too

FEATURE FLAG TEST
```
mutation {
  addFeatureFlag(email: "f@e45.co", flag: retro) {
    result
    error {
      message
    }
  }
}
```
- [ ] add the above to graphiql, but substitute in your own email. see retro meetings become enabled without a refresh.

bug burdown:
- [ ] bad styles
- [ ] can progress without at least 1 reflection

ackernaut commented:
@mattkrick hello there, I’m stuck trying to run a retro. I get re-routed to My Dashboard when hitting the retro meeting route. Left more info in Slack. Help when ya can, plz?

mattkrick commented:
@ackernaut tests are now passing, see if this is a workable branch

ackernaut commented:
ah, thanks @mattkrick I’ll give it a look. I still need to migrate down as well.

ackernaut commented:
I pulled latest and migrated down twice: `20180320200910 fix-retroReflection` and `20180320152455 voteSettings` and still having an issue, did I migrate down 1 too many?

jordanh commented:
@mattkrick looking at this now

jordanh commented:
# Test Cycle Template
## Status: in-progress/completed

## Tests
### VOTING TEST
- [ ] Go to the voting phase, can expand groups, but cannot drag cards
- [ ] can click the gray checkmark up to 3 times
- [ ] can remove votes by clicking pink checkmarks
- [ ] see updated vote info in the sidebar
- [ ] Browser B (different user) sees updated team votes remaining, but receives no user-specific info
- [ ] groups with only 1 reflection has an auto-generated title

### DISCUSS TEST
- [ ] can click the discuss topic in the sidebar & navigate to that stage
- [ ] can add tasks
- [ ] team members can see the editing state & card updates & card deletions


### SUMMARY TEST
- [ ] can click end meeting any time during the discuss phase
- [ ] when the meeting has ended, you go to the meeting summary
- [ ] the meeting summary shows you the tasks created, upvotied topics (does NOT distinguish between topics discussed/not discussed)
- [ ] you get an email with the summary there, too

### FEATURE FLAG TEST
```
mutation {
  addFeatureFlag(email: "f@e45.co", flag: retro) {
    result
    error {
      message
    }
  }
}
```
- [ ] add the above to graphiql, but substitute in your own email. see retro meetings become enabled without a refresh.

## Test Notes

jordanh commented:
# Test Cycle 1
## Status: in-progress

## Tests
### VOTING TEST
- [ ] Go to the voting phase, can expand groups, but cannot drag cards
- [ ] can click the gray checkmark up to 3 times
- [ ] can remove votes by clicking pink checkmarks
- [ ] see updated vote info in the sidebar
- [ ] Browser B (different user) sees updated team votes remaining, but receives no user-specific info
- [ ] groups with only 1 reflection has an auto-generated title

### DISCUSS TEST
- [ ] can click the discuss topic in the sidebar & navigate to that stage
- [ ] can add tasks
- [ ] team members can see the editing state & card updates & card deletions


### SUMMARY TEST
- [ ] can click end meeting any time during the discuss phase
- [ ] when the meeting has ended, you go to the meeting summary
- [ ] the meeting summary shows you the tasks created, upvotied topics (does NOT distinguish between topics discussed/not discussed)
- [ ] you get an email with the summary there, too

### FEATURE FLAG TEST
```
mutation {
  addFeatureFlag(email: "f@e45.co", flag: retro) {
    result
    error {
      message
    }
  }
}
```
- [ ] add the above to graphiql, but substitute in your own email. see retro meetings become enabled without a refresh.

## Test Notes

jordanh commented:
@mattkrick first test cycle complete. This is _exciting_! Only caught a few things.

mattkrick commented on src/server/graphql/mutations/helpers/endMeeting/sendNewMeetingSummary.js:
```diff
@@ -0,0 +1,55 @@
+import sendEmailPromise from 'server/email/sendEmail';
+import {RETROSPECTIVE} from 'universal/utils/constants';
+
+const getRetroMeeting = (meetingId) => {
```
i'm pretty sure i forgot how to write sql. not even mad.

mattkrick commented:
> Intermittent input bug when creating reflection

this is a problem with draft, it happens if you hit backspace on an empty card. i spent 10 mins trying to troubleshoot it and it wasn't an easy fix, so i'd say it's gonna be a half day thing.

> Voting titles

I can see value in folks picking the title they want for a single card. I don't think this should happen during the voting phase because it has nothing to do with voting & everything to do with grouping. I think this is best solved by making the title appear & be editable during the grouping phase. we ruled that out of the MVP, but can add it back in.

> Vote count

fixed

> Add task not working (intermittent)
Votes remaining occasionally won't update when fully spent (intermittent)


couldn't reproduce. let's test the heck out of it on a dev server

> Reflection and Task Outcome Card formatting not working:

fixed

> meeting cheat code not working

i've only had this happen after i hit a couple errors, and after a refresh it works again. if you have repro steps i'll be able to dive deeper.

jordanh commented:
> Intermittent input bug when creating reflection

>> this is a problem with draft, it happens if you hit backspace on an empty card. i spent 10 mins trying to troubleshoot it and it wasn't an easy fix, so i'd say it's gonna be a half day thing.

No worries. I'll open an issue now and we can prioritize it in the future.

> Voting titles

>> I can see value in folks picking the title they want for a single card. I don't think this should happen during the voting phase because it has nothing to do with voting & everything to do with grouping. I think this is best solved by making the title appear & be editable during the grouping phase. we ruled that out of the MVP, but can add it back in.

We'll let the user tension dictate our path here.

> Vote count

>> fixed

Sweet!

> Add task not working (intermittent)
> Votes remaining occasionally won't update when fully spent (intermittent)

>> couldn't reproduce. let's test the heck out of it on a dev server

Sounds good.

> Reflection and Task Outcome Card formatting not working:

> fixed

Sweet

> meeting cheat code not working

>> i've only had this happen after i hit a couple errors, and after a refresh it works again. if you have repro steps i'll be able to dive deeper.

Hmm, ok. I'll see if I can come up with reliable steps.

jordanh commented:
Let's merge this sucker and do some testing!
